party_menu/party_menu.lua

```diff
@@ -0,0 +1,521 @@
+
+            party_display_strings[i].name_hp_mp_tp = name_job_hp_format .. mp_format .. tp_format
+        elseif party[i] and party[i].zone_id ~= party[1].zone_id then
+            local name_zone_format_string = options.text_typeface .. ' ' .. tostring(options.text_size) .. 'px bold white'
+            local name_zone = get_name(i) .. 'Not in zone'
+            local name_zone_format = string.format('[' .. name_zone .. ']{' .. name_zone_format_string .. '}')
+
+            party_display_strings[i].name_hp_mp_tp = name_zone_format
+        else
+            party_display_strings[i].name_hp_mp_tp = ''
+        end
+    end
+end
+update_party_name_hp_mp_tp_strings('both')
+
+local function update_party_debuff_strings(update_type)
+    local start = 1
+    local end_point = 1
+    if update_type == 'player_only' then
+        start, end_point = 1, 1
+    elseif update_type == 'party_only' then
+        start, end_point = 2, 6
+    elseif update_type == 'both' then
+        start, end_point = 1, 6
+    end
+
+    for i = start, end_point do
+        if party[i] and party[i].zone_id == party[1].zone_id then
+            local debuffs_format_string = options.text_typeface .. ' ' .. tostring(options.text_size) .. 'px bold white'
+            local debuffs_format = string.format('[' .. get_debuff_string(i) .. ']{' .. debuffs_format_string .. '}')
+            party_display_strings[i].debuffs = debuffs_format
+        else
+            party_display_strings[i].debuffs = ''
+        end
+    end
+end
+update_party_debuff_strings('both')
+
+local party_window_states = {}
+
+local function create_party_window_states()
+    for i = 1, 6 do
+        party_window_states[i] = {
+            style = 'chromeless',
+            x = options.global_window_state.x + outer_padding,
+            width = options.global_window_state.width - 2 * outer_padding,
+            height = row_window_height,
+            y = options.global_window_state.y + i * outer_padding + (i - 1) * row_window_height,
+        }
+    end
+end
+create_party_window_states()
+
+local function update_ui_dimensions()
+    row_window_height = options.text_size + upper_inner_padding + lower_inner_padding
+    global_window_state.y = options.global_window_state.y - (get_party_size() - 1) * (row_window_height + outer_padding)
+    global_window_state.x = options.global_window_state.x
+    global_window_state.width = options.global_window_state.width
+    global_window_state.height = get_party_size() * (row_window_height + outer_padding) + outer_padding
+    for i, row_window in ipairs(party_window_states) do
+        row_window.x = options.global_window_state.x + outer_padding
+        row_window.width = options.global_window_state.width - 2 * outer_padding
+        row_window.y = global_window_state.y + outer_padding * i + row_window_height * (i - 1)
+        row_window.height = row_window_height
+    end
+end
+
+local function get_row_string()
```

Comment:
You call various functions in here with no arguments, which would result in `party[nil].something` in various places. Are you sure that works right?

Reply:
Luckily I don't think this function was actually ever called, it has been removed!